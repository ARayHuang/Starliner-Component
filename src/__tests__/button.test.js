import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../components/button';

describe('Button', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-button-className',	
		};

		const wrapper = shallow(
			<Button {...props}>
				<div>mock-children</div>
			</Button>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-button-className',
			variant: 'ghost',
			size: 'sm',
			isDisabled: false,
			text: 'mock-text',
			icon: <mock-icon />,
			onClick: () => {}		
		};
		
		const wrapper = mount(
			<Button {...props}>
				mock-text
			</Button>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().variant).toEqual(props.variant);
		expect(wrapper.props().size).toEqual(props.size);
		expect(wrapper.props().isDisabled).toEqual(props.isDisabled);
		expect(wrapper.props().icon).toEqual(props.icon);
		expect(wrapper.props().children).toEqual(props.text);
	})
})
