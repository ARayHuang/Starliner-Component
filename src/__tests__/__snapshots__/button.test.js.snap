// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<Fragment>
  <Button
    className="mock-button-className"
    size="sm"
    sx={
      Object {
        "border": "1px solid transparent",
        "borderRadius": "4px",
        "boxShadow": "0px 3px 1px -2px rgba(#000, 0.2), 0px 2px 2px 0px rgba(#000, 0.14), 0px 1px 5px 0px rgba(#000, 0.12)",
      }
    }
  />
</Fragment>
`;
